core/src/main/java/org/apache/gravitino/listener/api/event/CreateFilesetPreEvent.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.listener.api.event;
+
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.annotation.DeveloperApi;
+import org.apache.gravitino.listener.api.info.FilesetInfo;
+
+/** Represents an event triggered before creating a fileset. */
+@DeveloperApi
+public class CreateFilesetPreEvent extends FilesetPreEvent {
+  private final FilesetInfo createFilesetRequest;
+
+  public CreateFilesetPreEvent(
+      String user, NameIdentifier identifier, FilesetInfo createFilesetRequest) {
+    super(user, identifier);
+    this.createFilesetRequest = createFilesetRequest;
+  }
+
+  /**
+   * Retrieves the create fileset request.
+   *
+   * @return A {@link FilesetInfo} instance encapsulating the comprehensive details of create
+   *     fileset request.
+   */
+  public FilesetInfo createFilesetRequest() {
+    return createFilesetRequest;
```

Comment:
What you are returning here is not the full set of parameters for a CreateFileset request, so the info you retrieved should be a filesetInfo as the return type of the function indicates.

```suggestion
    return fileInfo;
```

Reply:
I'd like to keep the original naming, for it's more meaningful to me.